packages/svelte/src/internal/client/dom/blocks/svelte-head.js

```diff
@@ -19,12 +28,31 @@ export function head(render_fn) {
 	if (hydrating) {
 		previous_hydrate_nodes = hydrate_nodes;
 
-		let anchor = /** @type {import('#client').TemplateNode} */ (document.head.firstChild);
-		while (anchor.nodeType !== 8 || /** @type {Comment} */ (anchor).data !== HYDRATION_START) {
-			anchor = /** @type {import('#client').TemplateNode} */ (anchor.nextSibling);
+		// There might be multiple head blocks in our app, so we need to account for each one needing independent hydration.
+		if (head_anchor === undefined) {
+			head_anchor = /** @type {import('#client').TemplateNode} */ (document.head.firstChild);
+		}
+
+		var depth = 0;
+
+		while (head_anchor !== null) {
+			if (head_anchor.nodeType === 8) {
+				var data = /** @type {Comment} */ (head_anchor).data;
+				if (data === HYDRATION_START) {
+					if (depth === 0) {
+						break;
+					}
+					depth++;
+				}
+				if (data === HYDRATION_END) {
+					depth--;
+				}
+			}
+			head_anchor = /** @type {import('#client').TemplateNode} */ (head_anchor.nextSibling);
```

Comment:
If I change this to this
```suggestion
		while (
			head_anchor &&
			(head_anchor.nodeType !== 8 || /** @type {Comment} */ (head_anchor).data !== HYDRATION_START)
		) {
			head_anchor = /** @type {import('#client').TemplateNode} */ (head_anchor.nextSibling);
```
(basically what was there before)
all tests still pass. Why is it necessary to search for the head anchor like that?

Reply:
I thought that it would stop on inner anchors where there's an if statement or something similar. Maybe not, I'll change that back.